packages/compile-solidity/src/compilerSupplier/loadingStrategies/VersionRange.ts

```diff
@@ -52,10 +52,27 @@ export class VersionRange {
     }
   }
 
-  async list() {
-    const data = await this.getSolcVersions();
+  async list(index = 0) {
+    if (index >= this.config.compilerRoots.length) {
+      throw new Error(
+        `Failed to fetch the list of Solidity compilers from the following ` +
+          `sources: ${this.config.compilerRoots}. Make sure you are connected ` +
+          `to the internet.`
+      );
+    }
+    let data;
+    try {
+      data = await this.getSolcVersionsForSource(
+        this.config.compilerRoots[index],
+        index + 1
```

Comment:
So, I was really confused here why you were adding 1 to the `index` in cases when you weren't retrying.  This had me really confused.  Eventually I realized it was because `getSolcVersionsForSource` only uses that number for printing.

I think this is confusing and it would be better to *not* add 1 to the index here, but rather instead to add it at the last minute, when emitting the event, since it's purely a display concern (like happens on line 163).

There are several places where you do this; I'll mark them for convenience.

Reply:
So there is an `attemptNumber` variable in `getSolcVersionsForSource`. I don't think it would make sense to just pass the index as the second argument. What if I set `const attemptNumber = index + 1;` before using it as an argument? That would make it clear what is going on eh?

Reply:
Oh yeah, that also works... or just add a comment, honestly.